### ISSUE TITLE
apoc.export.csv.query() config for quotes: 'ifNeeded' produces invalid quotation

### ISSUE THREAD

**Issue by JanAntoniak** [neo4j-contrib/neo4j-apoc-procedures#4120]
## Expected Behavior (Mandatory)

apoc.export.csv.query with {quotes:'ifNeeded'} when adding quotation marks should do it in the same way as the call with {'always'}.

When a string contains a quotation mark it should be escaped (additional quotation mark added right before it).

That should be the output:

For nodes with name IN [ "\"Jan\" Apoc User", "\"Jan\" Apoc\"\" User" ] it should give the following output:

```

id:ID(Test),:Label,name

1,Test,"""Jan"" Apoc User"

2,Test,"""Jan"" Apoc"""" User"

```

 

## Actual Behavior (Mandatory)

apoc.export.csv.query with {quotes:'ifNeeded'} gives such output:

 

```

id:ID(Test),:Label,name

1,Test,""Jan" Apoc User"

2,Test,""Jan" Apoc"" User"

```

 

### Simple Dataset (where it's possibile)

 

```

CREATE (test:Test{id: 1, name:"\"Jan\" Apoc User"})

CREATE (test:Test{id: 2, name:"\"Jan\" Apoc\"\" User"})

```

 

 

### Steps (Mandatory)

 

  1. create test nodes with property name which contains quotation marks

  2. run the export with quotation option 'always'

```

call apoc.export.csv.query("MATCH (n:Test) RETURN n.id as `id:ID(Test)`, apoc.text.join(labels(n), ';') as `:Label`, n.name as name", "nodes/TestAlways.csv", {quotes:'always'})

YIELD file

RETURN file

```

  3. run the export with quotation option 'always' and 'ifNeeded'

```

call apoc.export.csv.query("MATCH (n:Test) RETURN n.id as `id:ID(Test)`, apoc.text.join(labels(n), ';') as `:Label`, n.name as name", "nodes/TestIfNeeded.csv", {quotes:'ifNeeded'})

YIELD file

RETURN file

```

  4. Compare the output files (TestAlways.csv and TestIfNeeded.csv) - value of property 'name' is treated differently in those two cases. 'neo4j-amin tool import'  accepts only the way that export with option 'always'  produces. 

 

## Specifications (Mandatory)

 

Currently used versions

 

### Versions

 

  - OS: both Windows and Linux

  - Neo4j: 4.4.15 (Enterprise) 

  - Neo4j-Apoc: 4.4.0.9

**Comment by Lojjs** [comment]
@JanAntoniak This sounds like a bug to me. I will track it internally and we will reach out to you once it is fixed.

Best regards Louise, Neo4j

**Comment by JoelBergstrand** [comment]
@JanAntoniak, thank you again for reporting the issue. We have implemented a fix that will be included in the next APOC 4.4 release. 

Kind regards 
Joel, Neo4j
